src/main/java/com/mongodb/hibernate/jdbc/MongoArray.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+public final class MongoArray implements ArrayAdapter {
+    private final Object contents;
+
+    public MongoArray(Object contents) {
+        this.contents = contents;
+    }
+
+    @Override
+    public Object getArray() {
+        // Hibernate ORM does not call `Connection.getTypeMap`/`setTypeMap`, therefore we are free to ignore it
```

Comment:
the above comment is not necessary from my perspective, for we know we are not developing JDBC Driver, and it is easy to know from `MongoConnection` implementation (no typeMap related method overriding) that we don't need to be concerned about type map.

For tech-savvy developer who is versed in Hibernate code logic, array is a relatively new feature and it is so natural that only basic features are implemented. Maybe in the long future type map is to be considered, but it is highly doubtable.

Reply:
> the above comment is not necessary

Our implementation of the `java.sql.Array.getArray` method violates its specification. The comment explains why such a violation is fine. I think, this comment is needed.

> we know we are not developing JDBC Driver

Yet we try to follow the specification where possible. If Hibernate ORM used `Connection.getTypeMap`/`setTypeMap`, we would likely have to take those into account. 

"we are not developing JDBC Driver" is not the reason we are diverging from the specification when implementing `java.sql.Array.getArray`. The reason is that Hibernate ORM does not use the methods mentioned in the comment. And this explanation worth to be put in a comment.